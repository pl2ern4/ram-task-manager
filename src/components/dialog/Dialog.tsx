import {useEffect, useState} from 'react';
import Dialog from '@mui/material/Dialog';
import DialogTitle from '@mui/material/DialogTitle';

type Props = {
    openDialog:boolean;
    title?: string;
    onClose: (args:boolean)=>void;
    children: React.ReactElement;
}
export default function FormDialog({openDialog,  onClose, title, children}: Props) {

  const [open, setOpen] = useState(openDialog)
  const handleClose = () => {
    setOpen(false);
    onClose(false)
  };

  useEffect(()=>
    setOpen(openDialog)
  ,[openDialog])

  return (
      <Dialog
        open={open}
        onClose={handleClose}
        scroll="paper"
        aria-labelledby="scroll-dialog-title"
        aria-describedby="scroll-dialog-description"
      >
        {title && <DialogTitle>{title}</DialogTitle>}
        {children}
      </Dialog>
  );
}