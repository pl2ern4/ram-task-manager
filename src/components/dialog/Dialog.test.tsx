import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import Dialog from './Dialog';

describe('Dialog', () => {
    it('should Match Snapshot', async () => {
        const mockFn = jest.fn();
        const { container } = render(<Dialog onClose={mockFn} openDialog={true}><div>Test</div></Dialog>)
        expect(container).toMatchSnapshot();
    });
    it('should Match Snapshot, when dialog state `openDialog` set to false', async () => {
        const mockFn = jest.fn();
        const { container } = render(<Dialog onClose={mockFn} openDialog={false}><div>Test</div></Dialog>)
        expect(container).toMatchSnapshot();
    });
    it('should Match Snapshot, when title state set `title` set to null', async () => {
        const mockFn = jest.fn();
        const { container } = render(<Dialog title="test" onClose={mockFn} openDialog={true}><div>Test</div></Dialog>)
        expect(container).toMatchSnapshot();
    });
})