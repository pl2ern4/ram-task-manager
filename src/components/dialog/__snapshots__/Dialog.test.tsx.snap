// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog should Match Snapshot 1`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Dialog should Match Snapshot, when dialog state \`openDialog\` set to false 1`] = `<div />`;

exports[`Dialog should Match Snapshot, when title state set \`title\` set to null 1`] = `
<div
  aria-hidden="true"
/>
`;
